app/controllers/rentalsandrentaroom/expenses/RentalsAndRaRExpensesCheckYourAnswersController.scala

```diff
@@ -100,4 +83,60 @@ class RentalsAndRaRExpensesCheckYourAnswersController @Inject() (
       PropertyBusinessTravelCostsSummary.row(taxYear, request.userAnswers, RentalsRentARoom),
       OtherAllowablePropertyExpensesSummary.row(taxYear, request.userAnswers, RentalsRentARoom)
     )
+
+  def onSubmit(taxYear: Int): Action[AnyContent] = (identify andThen getData andThen requireData).async {
+    implicit request =>
+      val context =
+        JourneyContext(taxYear, request.user.mtditid, request.user.nino, "rentals-and-rent-a-room-expenses")
+
+      request.userAnswers.get(RentalsAndRentARoomExpenses) match {
+        case Some(rentalsRentARoomExpenses) =>
+          propertySubmissionService
+            .saveJourneyAnswers(context, rentalsRentARoomExpenses)
+            .flatMap {
+              case Right(_) =>
+                auditExpensesCYA(taxYear, request, rentalsRentARoomExpenses, isFailed = false)
+                Future.successful(Redirect(routes.RentalsRaRExpensesCompleteController.onPageLoad(taxYear)))
+              case Left(_) =>
+                auditExpensesCYA(taxYear, request, rentalsRentARoomExpenses, isFailed = true)
+                Future.failed(InternalErrorFailure("Failed to save Rentals and Rent a Room Expenses section."))
+            }
+
+        case None =>
+          logger.error("RentalsAndRentARoomExpenses section is not present in userAnswers")
+          Future.failed(InternalErrorFailure("RentalsAndRentARoomExpenses section is not present in userAnswers"))
+      }
+
+  }
+
+  private def auditExpensesCYA(
+    taxYear: Int,
+    request: DataRequest[AnyContent],
+    rentalsAndRentARoomExpenses: RentalsAndRentARoomExpenses,
+    isFailed: Boolean
+  )(implicit
+    hc: HeaderCarrier
+  ): Future[Unit] =
+    businessService
+      .getUkPropertyDetails(request.user.nino, request.user.mtditid)
```

Comment:
@vivekhmrc I hope you remember our conversation during the refactoring session. You agreed to fix the redundant business details API call. If we don't address it now, others might start copying and pasting, which will make it much harder to refactor later on.